### DISCUSSION TITLE
What features would you like to see?

### DISCUSSION THREAD

**Discussion by markseuffert** [datenstrom/community#991] [Ask a question] [answered]
Hello everybody out there using Datenstrom Yellow. We are moving towards a 1.0 version and it would be great to collect more ideas. I think the application feels lightweight and fast. Although it is designed for small websites, some people use it with hundreds of pages. What would you like to see? Anything not working or missing that you need? What do you like/dislike?

**Comment by schulle4u** [comment]
There have been a few ideas on Discord and in previous discussions that might be worth mentioning here again. Installing extensions in the browser has been talked about as a Christmas present for years, right? 😁

We also had a bit more discussion about semantic HTML. Personally, I would like to see default layouts use more semantic HTML elements instead of nesting everything in div elements. Div elements are still useful and even needed for extensions, but the basic structure of a website can be designed just as well with semantic elements such as `<header>`, `<nav>`, `<main>` or `<footer>`. Of course, there is no reason not to use the existing CSS classes to keep all themes compatible. After all these elements make the code more readable, improve accessibility, and, last but not least, should make it easier to port existing CSS frameworks/themes to Yellow.

**Comment by PetersOtto** [comment]
I am actually very happy with Yellow. If life were a musical request program, I would wish for three things.

1. at the moment you can only update all or no extensions via the Edit extension. It would be nice if you could make a selection here.

2. at the moment only official extensions can be updated via the update extension. It would be nice if you could also do this with your own extensions.

3. easy organization (replace, move, delete ...) of the media files in the browser would be nice.

&#128512;

**Comment by xrat** [comment]
I, too, am very happy with Yellow.

Things, hacks and other features I frequently add by changing the code or including them in my personal extension/plugin are:

* Extend Markdown syntax
* Add a keyboard shortcut to edit a page (I use Ctrl-E)
* Add my own `custom.css` for minor mods of the public Yellow themes I use
* Make TOC collapsible
* Log login fails and successes (in my case to syslog)

One bigger issue I have is related to how Yellow changes filenames and paths of pages. While I understand why Yellow does rename files, it happened to me too often that links broke because pages got renamed without anybody noticing at the time it happened and when it could have been quickly reverted. However, I have not yet come up with a good alternative.

One more specific issue: Does the regex in `normaliseToken`, i.e. `/^[\d\-\_\.]+(.*)$/` need to be so broad? For instance, I couldn't have a page named "2fa" or "2factorauthentication".

**Comment by pftnhr** [comment]
I would be very happy if the edit panel could be better designed so that I can ensure that all the buttons that are important to me are also accessible in portrait mode on mobile. And maybe an option for inline editing when I'm logged in.

**Reply by pftnhr** [reply]
I would like to be a little more precise:

By inline editing I mean something like double clicking on a page or post and you can edit the content directly in the page or post without having to click on "Edit" at the top. As it currently stands, it only works well on a computer. If you want to maintain content on a mobile device, it is extremely cumbersome, especially if you want to add images.

And I would be happy if the editor could be customised via a template that is then also located in the 'layouts' directory, for example.

**Comment by GiovanniSalmeri** [comment]
Yellow 1.0 is great news! I too am very happy with Yellow and actually I use it also for not-so-small sites ([Dialegesthai](https://mondodomani.org/dialegesthai/) has almost 1200 pages and 60 MB of text) and it works perfectly also for them. Kudos to the developers for the wonderful work!

As regards ideas for anything missing: I think that the core (with the core extensions) is excellent as it is. Perhaps a file manager (see https://github.com/datenstrom/yellow/discussions/66 and https://github.com/datenstrom/yellow/discussions/130) would be a much welcomed complement to Edit for those who prefer to modify a site in their browser instead of using a text editor + FTP.

All the other things that come to my mind belong in the extensions. Perhaps it would be useful to maintain a public list of ideas for new extensions?

**Comment by auer-kk** [comment]
I would like to see a revised sorting/grouping of pages (by title/tag) that works with special characters like (german) umlauts.
Example: Using shortcuts like `[wikipages]/[wikitags]` or calling `$pages->group(…)` right now sorts those umlauts to the end
`[Ax, Bx, Cx, Äx]`. My expected/wanted sort order would be `[Ax, Äx, Bx, Cx]` instead.

**Comment by Brotkasten-a11y** [comment]
A fix for a general issue with markdown parsing would be nice: If you write an email address with a longer domain than 3 charakters like a '.vision' domain (e.g., info@hello.vision), the markdown extension generates a link like [info@hello.vis](info@hello.vis)ion, leaving the 'ion' part outside the link.

**Reply by markseuffert** [reply]
Good idea, I updated the [Markdown extension](https://github.com/annaesvensson/yellow-markdown). The latest version now supports emails with long top-level domain (TLD).

**Comment by ghost** [comment]
It would be very cool a plugin for events, with a calendar. Like those we see in the libraries websites.

**Reply by GiovanniSalmeri** [reply]
You mean something like [Diary](https://github.com/GiovanniSalmeri/yellow-diary)? Buried in my HD there is also a version of it which integrates [FullCalendar](https://fullcalendar.io/), I could publish it if it is useful.

**Reply by ghost** [reply]
This plugin is marvelous ❤️❤️❤️
I will download It, thank you!

**Comment by pftnhr** [comment]
I would also like to be able to create new posts via CLI using the `page-new-*.md` templates. Maybe `php yellow.php create post` or `php yellow.php create page` or something like that which opens whatever editor your shell is set to.

**Comment by glutfuchs** [comment]
There is a relatively new mastodon backlink feature in Mastodon which would be nice.
https://blog.joinmastodon.org/2024/07/highlighting-journalism-on-mastodon/
The code looks something like this.
`<meta name="fediverse:creator" content="...profile descriptor here..." />`
It is possible to edit the header.html but it would be nice to control the author/creator seperately for every article.

**Reply by schulle4u** [reply]
Something like this? 

```
<meta name="fediverse:creator" content="<?php echo $this->yellow->page->getHtml("metaFediverseHandle") ?>" />
```

Now you should be able to add `MetaFediverseHandle: @someone@mastodon.social` to your page settings. 🙂

**Reply by glutfuchs** [reply]
Yes, thank you so much 🙏 
A lot easier as I thought.

**Comment by pftnhr** [comment]
I like the idea of Markdown render hooks, such as those offered by Hugo for, e.g. images and links: https://gohugo.io/render-hooks/

Something like this makes one or the other extension obsolete and you no longer have to wait/hope for an extension developer to restore compatibility when updating your Yellow installation.

**Comment by caius-martinus** [comment]
I first tried Yellow in 2014 or 2015. One thing I still miss is to have the default theme to be responsive and mobile friendly : hamburger menu when small (possibly with one level of submenu), regular menu when larger view.

**Comment by henryVr7** [comment]
Great question! I'd love to see Mastodon integrated so I can, for example, see all the comments on a blog post published there.
Besides, Yellow desperately needs proper media management.

**Comment by markseuffert** [comment, marked as answer]
Thanks everyone for your ideas. Hope we can add some of your ideas to the next release, for example the support for installing extensions in a web browser. If something is missing that hasn't been mentioned, please open a new discussion and consider contributing with code.
